packages/api/lib/errors.js

```diff
@@ -0,0 +1,46 @@
+
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+class InvalidParameterError extends Error {
+  constructor (message) {
+    super(message);
+    this.name = this.constructor.name;
+    Error.captureStackTrace(this, this.constructor);
+  }
+}
+class ResourceNotFoundError extends Error {
+  constructor (message) {
+    super(message);
+    this.name = this.constructor.name;
+    Error.captureStackTrace(this, this.constructor);
+  }
+}
+
+function handleError (res, err) {
+  if (err instanceof InvalidParameterError) {
+    res.status(400).send({ error: err.message, name: err.name });
+  } else if (err instanceof ResourceNotFoundError) {
+    res.status(404).send({ error: err.message, name: err.name });
+  } else {
+    console.error(err.stack);
+    res.status(500).send({ error: err.message, name: err.name });
```

Comment:
what I've seen done, and we might want to move to eventually, is that rather than serializing `err.message`, we add meta information closer to the call site of the exception: using our own error class, so that we know we've controlled the value of `message` and `code`.

* if we don't get a sanitized error object, we send a default `500`, with a message like `unknown error`.
* if we have the custom error object (which we can check for using `instanceof`), decorated at the call site, we throw the custom error.

For now, I would send:

```
400 "bad request"
404 "not found"
500 "unknown error"
```

Rather than `err.message`, as this also has the potential to leak information to the user. If we want better error messages for the client side, we can use the approach described above, of a custom error object

Reply:
Changed to only give that information.